switchboard/sbdut.py

```diff
@@ -614,10 +616,43 @@ def simulate(
                     args=plusargs_to_args(plusargs) + args
                 )
 
+        # Add newly created Popen object to subprocess list
+        self.subprocess_list.append(p)
+
         # return a Popen object that one can wait() on
 
         return p
 
+    def terminate(
+        self,
+        stop_timeout=10,
+        use_sigint=False
+    ):
+        if not self.subprocess_list:
+            raise Exception('No ongoing simulation.'
+                'Please call simulate before trying to terminate.')
```

Comment:
I think you can just return, raising an exception is overkill.

Reply:
Done.